erpnext-openmrs/src/main/java/com/ozonehis/eip/erpnext/openmrs/handlers/AddressHandler.java

```diff
@@ -30,25 +30,33 @@ public class AddressHandler {
     @Autowired
     private FrappeClient frappeClient;
 
-    public boolean addressExists(String addressName) {
+    /**
+     *  Get an address by name
+     *
+     * @param addressName the name of the address
+     * @return the address if it exists, empty otherwise
+     */
+    public Optional<Address> getAddress(String addressName) {
         try (FrappeResponse response = frappeClient.get("Address", addressName).execute()) {
-            if (response.code() == HttpStatus.SC_OK) {
-                TypeReference<FrappeSingularDataWrapper<com.ozonehis.eip.model.erpnext.Address>> typeReference =
-                        new TypeReference<>() {};
+            switch (response.code()) {
+                case HttpStatus.SC_NOT_FOUND:
+                    return Optional.empty();
+                case HttpStatus.SC_OK:
+                    TypeReference<FrappeSingularDataWrapper<Address>> typeReference = new TypeReference<>() {};
 
-                FrappeSingularDataWrapper<com.ozonehis.eip.model.erpnext.Address> addressWrapper =
-                        response.returnAs(typeReference);
-                return addressWrapper.getData().getAddressName().equals(addressName);
-            } else {
-                return false;
+                    FrappeSingularDataWrapper<Address> addressWrapper = response.returnAs(typeReference);
+                    return Optional.of(addressWrapper.getData());
+                default:
+                    throw new FrappeClientException(
+                            "Error while checking if address with uuid: " + addressName + " exists");
```

Comment:
Does `addressName` stores a UUID?

Reply:
It's `UUID` but the field name is `name`. For consistent I'll renamed to field name.